types/instance.go

```diff
@@ -70,3 +80,75 @@ func (i Instance) GetItem() interface{} {
 func (i Instance) GetType() string {
 	return "instance"
 }
+
+type JsonResponseBody struct {
+	Json map[string]interface{}
+}
+
+type RemoteResponse struct {
+	Code  int
+	Body  string
+	Json  JsonResponseBody
+	Error bool
```

Comment:
I don't see the benefit of mixing error with response. Could you please convert error as a plain return value? I think `Code` is enough to determine the success of the request.